ngraph_bridge/enable_variable_ops/ngraph_rewrite_for_tracking.cc

```diff
@@ -122,10 +125,65 @@ Status RewriteForTracking(Graph* graph, int graph_id) {
 
     }  // end of checking if it is NGVariableType
   }    // end of looping through the nodes in the graph
+
   for (auto node : replaced_nodes) {
     graph->RemoveNode(node);
   }
 
+  for (auto node : add_sync_nodes_to) {
+    // Since the node takes in variable as a reference
+    // and is not supported by NGraph, it might update the
+    // variable hence the sync node is required here.
+    for (auto edge : node->in_edges()) {
+      if (IsRefType(node->input_type(edge->dst_input()))) {
+        NodeBuilder::NodeOut input_ref;
+        input_ref = NodeBuilder::NodeOut(edge->src(), edge->src_output());
+        DataType dtype;
+        TF_RETURN_IF_ERROR(GetNodeAttr(edge->src()->attrs(), "dtype", &dtype));
+        string shared_name;
+        TF_RETURN_IF_ERROR(
+            GetNodeAttr(edge->src()->attrs(), "shared_name", &shared_name));
+        string sync_node_name = edge->src()->name() + "/sync_node";
+        Node* sync_node;
+        NodeBuilder nb =
+            NodeBuilder(sync_node_name, "NGraphVariableUpdateNGTensor")
+                .Input(input_ref)
+                .Attr("ngraph_graph_id", graph_id)
+                .Attr("ngraph_variable_shared_name", shared_name)
+                .Attr("T", dtype)
+                .Device(edge->src()->assigned_device_name());
+        Status status = nb.Finalize(graph, &sync_node);
+        TF_RETURN_IF_ERROR(status);
+        sync_node->set_assigned_device_name(
+            edge->src()->assigned_device_name());
+
+        // If the input edge is going to index 0 of the TF op,
+        // then move output edges from the TF op
+        // since that is the output we want.
+        if (edge->dst_input() == 0) {
+          // Also check if the output of the TF op is a ref type and only
+          // move the output edges then otherwise keep the output edges
+          // attached to the TF op and add a control edge from the sync
+          // node to the output/s of the TF op.
+          if (IsRefType(node->output_type(0))) {
+            TF_RETURN_IF_ERROR(ReplaceOutputEdges(graph, node, sync_node));
+          } else {
+            for (auto out_edge : node->out_edges()) {
+              graph->AddEdge(sync_node, Graph::kControlSlot, out_edge->dst(),
+                             Graph::kControlSlot);
+            }
+          }
+        }
+
```

Comment:
Note to self:

in case of IsVariableInitialized:

`Var--(ref edge)-->IsVariableInitialized--(nonref edge)-->X`


This will become:

```Var--(ref edge)-->IsVariableInitialized--(nonref edge)-->X
                                            |
                                            v
                                        sync
```